src/Aspire.Hosting/JavaScript/JavaScriptExtensions.cs

```diff
@@ -0,0 +1,88 @@
+    /// <summary>
+    /// Adds a JavaScript runtime to the application model. The runtime and dependencies should available on the PATH.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/> to add the resource to.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="scriptPath">The path to the script that the runtime will execute.</param>
+    /// <param name="workingDirectory">The working directory to use for the command. If null, the working directory of the current process is used.</param>
+    /// <param name="args">The arguments to pass to the command.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<JavaScriptAppResource> AddJavaScriptApp(this IDistributedApplicationBuilder builder, string name, string scriptPath = null, string? workingDirectory = null, string[]? args = null)
+    {   
+        scriptPath ??= "node";
+        args ??= [];
+        string[] effectiveArgs = [scriptPath, .. args];
+        workingDirectory ??= Path.GetDirectoryName(scriptPath)!;
+        workingDirectory = PathNormalizer.NormalizePathForCurrentPlatform(Path.Combine(builder.AppHostDirectory, workingDirectory));
+
+        var resource = new JavaScriptAppResource(name, scriptPath, workingDirectory, effectiveArgs);
+
+        return builder.AddResource(resource)
+                      .WithJavaScriptAppDefaults();
+    }
+
+    /// <summary>
+    /// Adds a node application to the application model. Executes the npm command with the specified script name.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/> to add the resource to.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="workingDirectory">The working directory to use for the command. If null, the working directory of the current process is used.</param>
+    /// <param name="scriptName">The npm script to execute. Defaults to "start".</param>
+    /// <param name="args">The arguments to pass to the command.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<JavaScriptAppResource> AddJavaScriptCLIApp(this IDistributedApplicationBuilder builder, string name, string workingDirectory, string? scriptName = null, string[]? args = null)
+
+    {   
+        string[] allArgs;
+
+        // Check if scriptName is "npm" and adjust allArgs accordingly
+        if (string.Equals(scriptName, "npm", StringComparison.OrdinalIgnoreCase))
+        {
+            allArgs = args is { Length: > 0 }
+                ? new string[] { "run", scriptName, "--" }.Concat(args).ToArray()
+                : new string[] { "run", scriptName };
+        }
+        else
+        {
+            // If scriptName is not npm, use developer config or an empty array
+            allArgs = args ?? Array.Empty<string>();
```

Comment:
Here we should throw an error and say. "Commands can't be empty" or something of the like